Documentation/kvstore.rst

```diff
@@ -4,6 +4,12 @@ Key-Value Store
 Cilium uses an external key-value store to exchange information across multiple
 Cilium instances:
 
+.. note::
+
+    Cilium does not support removing and recreating a kvstore that is currently in use.
```

Comment:
Hmm. Since the beginning, we often stated that the source of truth for kvstore state was in the agents, and therefore if you deleted the content of the KVStore, all agents would recreate the state and recover.

It sounds like at some point the IPSec feature specifically changed this assumption. Is that fair to say?

Reply:
Since we never stated that removing kvstore state is OK in the docs, and we probably aren't confident enough to make such a statement now, maybe the best approach is to word the statement roughly with the following points:

- Don't remove kvstore state manually unless you know exactly what you're doing.
- If you manually remove kvstore state when transparent encryption is enabled then that may cause permanent connectivity disruption. (for pods managed by Cilium? for nodes?)